### ISSUE TITLE
Adding boost

### ISSUE THREAD

**Issue by sandreialexandru** [pihome-shc/pihome#426]
When creating a zone with a maximum operation time greater than 127 minutes, boost does not get added and returns an error. I believe that this is caused by the type of "minute" column in boost table (tinyint). Also, I would suggest that when manually adding a boost we should use some limitation for the temperature and time like maximum zone temperature and time instead of hard coded values.

**Comment by sandreialexandru** [comment]
Maybe using sliders instead of dropdown lists is easier to implement.

**Comment by twa127** [comment]
I guess we have never seen this issue as max operation time have never been used up until now

**Comment by sandreialexandru** [comment]
Leaving out of discussion particular cases, I think that we should create some limitations on boost variables (time&temperature), for security concerns.

**Comment by pihome-shc** [comment]
i think for safety we should introduce zone running time limit as have temperature limit and use hysteresis_time from zone_sensor table, open for discussion on this one.

**Comment by sandreialexandru** [comment]
Having time and temperature limits for boost options is a way to set cut-off. Using hysteresis_time column from zone_sensor table as a maximum time limit for boost is a good idea, but it should be renamed. Doing it like this we won't have to modify the database structure.

**Comment by pihome-shc** [comment]
we already have boost max time in boost table why we need to save it twice? i agree on max zone temperature with slider would be good as zone will never go over its max zone temperature even if user define 400c for boost

**Comment by sandreialexandru** [comment]
It is like linking everything together. Why would someone be able to set a boost temperature to 400 if the max zone temperature is set at 25? The slider should go as up as the maximum temperature for that zone.

**Comment by sandreialexandru** [comment]
You are right that time should not be saved twice and maybe there should be just a text box where the user can input time manually, with just a simple html validation for numbers.

**Comment by pihome-shc** [comment]
but we already have time and temperature cutoff in boost table, i was talking about change from dropdown to slider and leave boost related data in boost table

**Comment by sandreialexandru** [comment]
I agree. The database would be more organized.
